GitVersionCore/BranchConfigurationCalculator.cs

```diff
@@ -117,8 +131,8 @@ public class BranchConfigurationCalculator
             else
                 errorMessage = "Failed to inherit Increment branch configuration, ended up with: " + string.Join(", ", possibleParents.Select(p => p.Name));
 
-            var hasDevelop = repository.Branches["develop"] != null;
-            var branchName = hasDevelop ? "develop" : "master";
+            var developBranch = repository.Branches.FirstOrDefault(b => Regex.IsMatch(b.Name, "^develop", RegexOptions.IgnoreCase));
```

Comment:
Why using regex for this one? Maybe it's even better to make this a setting (I see some people using "dev" instead on the inet).